README.md

```diff
@@ -1,3 +1,123 @@
+
+These components are at the foundation of [Falco](https://github.com/falcosecurity/falco) and other projects that work with the same kind of data.
+
+This component stack mainly operates on a data source: system calls. This data source is collected using either a kernel module or an eBPF probe, which we call *drivers*. On top of the drivers, libscap manages the data capture process, libsinsp enriches the data, and provides a rich set of API to consume the data. Furthermore, these two libraries also implement a [plugin](https://github.com/falcosecurity/plugins) framework that extends this stack to potentially any other data sources.
+## Project Layout
+
+* [_driver/_](./driver) contains kernel module and eBPF probe source code,
+so-called **drivers**.       
+* [_userspace/_](./userspace) contains libscap and libsinsp libraries code,
+plus chisels related code and common utilities.
+  * **libscap** (aka lib for *System CAPture*) is the userspace library
+  that directly communicates with the drivers, reading syscall events from
+  the ring buffer (where drivers place them), and forwarding them
+  up to libsinsp. Moreover, libscap implements OS state collection and
+  supports reading/writing to scap files.  
+  * **libsinsp** (aka lib for *System INSPection*) receives events from
+  libscap and enriches them with machine state: moreover, it performs
+  events filtering with rule evaluation through its internal rule engine.
+  Finally, it manages outputs. 
+  * **chisels** are just little Lua scripts to analyze an event stream
+  and perform useful actions. In this subfolder, the backend code for
+  chisels support can be found.  
+* [_proposals/_](./proposals) unexpectedly contains the list of proposals.
+* [_cmake/modules/_](./cmake/modules) contains modules to build
+external dependencies, plus the libscap and libsinsp ones; consumers
+(like Falco) use those modules to build the libs in their projects.
+
+## Build
+
+Libs relies upon `cmake` build system.  
+
+### Bundled deps
+
+Easiest way to build the project is to use BUNDLED_DEPS option, ie:
+```bash
+mkdir build && cd build
+cmake -DUSE_BUNDLED_DEPS=true ../
+make sinsp
+```
+> **NOTE:** take a break as this will take quite a bit of time (around 15 mins, dependent on the hardware obviously).
+
+### Use system deps
+
+To install using the system deps, first make sure to have all the needed packages installed.  
+Refer to https://falco.org/docs/getting-started/source/ for the list of dependencies.  
+
+Then, simply issue:
+```bash
+cmake ../
+make sinsp
+```
+
+> **NOTE:** using system libraries is useful to cut compile times down, as this way it will only build libs, and not all deps.  
+> On the other hand, system deps version may have an impact, and we cannot guarantee everything goes smoothly while using them.
```

Comment:
These parts are not super clear to me. My main question is what kind of output does this generate and in which cases I'll want to do this. While I really like the instructions for the other cases, that is if I `make` the kernel module I'll get a loadable kmod, if I compile the tests I'll be able to run the tests, etc... I don't really know why I would want to compile `sinsp` (+scap) by itself.